### PR TITLE
Fix UPE methods checkbox selection

### LINKED ISSUE #4629: Payment methods deselected
### Describe the bug
When quickly selecting more than one payment method, only the first one is selected.  And if there are selected methods already, quickly selecting the rest of the payment methods; only the last one will be selected.

### Screenshots
https://user-images.githubusercontent.com/56378160/186328647-6e2e0175-75f3-44a1-96e9-33f7dbc41320.mov

### To Reproduce
1. Go to WooCommerce -> settings -> payments
2. Find `Payments accepted on checkout` section
3. Click on the `enable in your store` button
4. Select at least two or three payment methods quickly
5. See only the first one is selected

### Actual behavior
When quickly selecting more than one payment method, only the first one is selected

### Expected behavior
When quickly selecting more than one payment method, all the selected payment methods should be selected

### PR DESCRIPTION
**Pull request by nagpai**
Fixes #4629 

#### Changes proposed in this Pull Request

The PR fixes the issue of UPE payment buttons getting unselected, in WC Pay settings, when multiple options are selected consecutively. 

##### Summary of the issue
The issue was happening due to a 1.5 second delay given for design reasons, to accomodate a possibility of a confirmation modal in certain cases. ( related to: paJDYF-2rk-p2 ). Multiple options clicked within the 1.5 second time gap would make only the last one checked, and would uncheck the previous ones. ( check video below )

**Video of before**

[Video of Before](https://user-images.githubusercontent.com/56378160/186328647-6e2e0175-75f3-44a1-96e9-33f7dbc41320.mov)

##### Summary of changes done

The function that managed [adding](https://github.com/Automattic/woocommerce-payments/blob/f5962d92e4b37ded818dcf1baf391a06cce943a0/client/payment-methods/index.js#L137) and [removing ](https://github.com/Automattic/woocommerce-payments/blob/develop/client/payment-methods/index.js#L144)payment methods from the `enabled_payment_methods_id` state array, took an outdated version of the complete state array, added the new payment method, and overwrote it fully. 

The change here makes sure that the associated actions take only the payment method ID that needs to be changed, and is either concatenated or removed from the most updated `enabled_payment_methods_id`  state array.  This makes sure that each check or uncheck action, affects only that checkbox ( payment method ) and not the rest. 

**After :**
Screen Capture on 2022-11-25 at 11-11-08

#### Testing instructions

* Go to WooCommerce -> Settings -> Payments
* Make sure you have UPE payment options enabled
* Find Payments accepted on checkout section
* Select at least two or three payment methods quickly
* Make sure that all the selected payment methods remain checked and do not roll back to unchecked
* Repeat the test with consecutively unchecking. You may not be able to uncheck multiple boxes quickly due to modals. 
* Test clicking okay or cancel in the removal confirmation modals. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️) - Updated existing tests
- [x] Tested on mobile (or does not apply) 

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by nagpai** [comment]
Additional comments: 

* Similar issue also seems to occur with the Payment method list selection that appears when we enable UPE for the first time. I intend to create a separate PR for this 
image


* TS migration seems larger effort and hence keeping it outside of the scope of this PR. 
* I would appreciate some experienced review and help with the test component. I have a deep feeling I need to do better there :D

**Review by tpaksu — APPROVED** [review]
You did a great job here, it's a clever solution! Thanks for working on this. Code reads good, understandable, variables are descriptive, tests pass, and tested it manually, works good. You can 🚢  it!

**Comment by tpaksu** [comment]
By the way, just remembered something, did you test if the UPE onboarding ones still work?

**Edit:** Still works but has the same bug. Why do you need TS migration to fix it? You can just fix it there.

**Comment by nagpai** [comment]
@tpaksu , Can't thank you enough for the swift review and check ( and also for all your help understanding the context of the issue earlier ) . The issue is thanks to my team and @vbelolapotkov 🙌 , who guided me.

I tested the UPE onboarding bit separately after applying this patch, and it does not seem to work. From a top-level look, It seems to use a different component that I will look into separately and fix. I have created an issue to [track it here](https://github.com/Automattic/woocommerce-payments/issues/5193). 

I mentioned the TypeScript part as a separate independent point, since we usually explore changing JS files to TS. If the effort is not much we include it in the same PR.

File: changelog/fix-4629-UPE-methods-selection [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Fix UPE checkboxes getting deselected
```
Comment (kalessil) [inline changelog/fix-4629-UPE-methods-selection]: Can we improve this entry to reflect it on the setting page and add some clarification - this is an external type of communication.

**Review by kalessil — APPROVED** [review]
LGTM! See also my note about the changelog entry.

File: changelog/fix-4629-UPE-methods-selection (hunk 1 above)
Comment (nagpai) [reply changelog/fix-4629-UPE-methods-selection]: Thanks, @kalessil - I have updated it to more clear and elaborate message here : https://github.com/Automattic/woocommerce-payments/pull/5171/commits/ce84518479ebbf901852ce5e47d770457f58e35b

Hope this is fine.

Comment (kalessil) [reply changelog/fix-4629-UPE-methods-selection]: Yay, that's much better! Thanks.

**Review by dpaun1985 — APPROVED** [review]
✅ LGTM!

**Pull request merged**